src/System.Numerics.Vectors/src/System/Numerics/Vector.cs

```diff
@@ -363,7 +363,8 @@ public unsafe Vector(T[] values, int index)
         {
             if (values == null)
             {
-                throw new ArgumentNullException("values");
+                // Match the JIT's exception type here. For perf, a NullReference is thrown instead of an ArgumentNull.
+                throw new NullReferenceException(SR.Arg_NullArgumentNullRef);
             }
             if (index < 0 || (values.Length - index) < Count)
```

Comment:
just noticed that if I was expecting to use these without SIMD, I'd definitely appreciate the perf more than telling me which of the args was the cause of NRE, if I happen to get one.

It is ok if that is not a targeted scenario.